phparkitect.php

```diff
@@ -14,27 +14,94 @@
 
 return static function (Config $config): void
 {
-    $srcClassSet = ClassSet::fromDir(__DIR__.'/src');
+    $specClassSet = ClassSet::fromDir(__DIR__ . '/src/Sylius/{Behat,Component/*,Bundle/*}/spec');
 
-    $rules = [];
+    $config->add(
+        $specClassSet,
+        Rule::allClasses()
+            ->that(new Extend(ObjectBehavior::class))
+            ->should(new HaveNameMatching('*Spec'))
+            ->because('This is a convention from PHPSpec')
+        ,
+        Rule::allClasses()
+            ->that(new Extend(ObjectBehavior::class))
+            ->should(new IsFinal())
+            ->because('Specifications should not be extendable')
+        ,
+    );
 
-    $rules[] = Rule::allClasses()
-        ->that(new ResideInOneOfTheseNamespaces('Sylius\Component'))
-        ->should(new NotDependsOnTheseNamespaces('Sylius\Bundle'))
-        ->because('Sylius components should be stand-alone')
-    ;
+    $testsClassSet = ClassSet::fromDir(__DIR__ . '{/tests,/src/Sylius/Bundle/*/Tests}');
 
-    $rules[] = Rule::allClasses()
-        ->that(new Extend(ObjectBehavior::class))
-        ->should(new HaveNameMatching('*Spec'))
-        ->because('This is a convention from PHPSpec')
-    ;
+    $config->add(
+        $testsClassSet,
+        Rule::allClasses()
+            ->that(new HaveNameMatching('*Test$'))
+            ->should(new IsFinal())
+            ->because('Tests should not be extendable')
+        ,
+    );
 
-    $rules[] = Rule::allClasses()
-        ->that(new Extend(ObjectBehavior::class))
-        ->should(new IsFinal())
-        ->because('Specifications should not be extendable')
-    ;
+    $separationClassSet = ClassSet::fromDir(__DIR__ . '/src/Sylius/{Component,Bundle}');
 
-    $config->add($srcClassSet, ...$rules);
+    $config->add(
+        $separationClassSet,
+        Rule::allClasses()
+            ->that(new ResideInOneOfTheseNamespaces('Sylius\Component'))
+            ->should(new NotDependsOnTheseNamespaces('Sylius\Bundle'))
+            ->because('Sylius components should be stand-alone')
+        ,
+        Rule::allClasses()
+            ->except('Sylius\Component\Core')
+            ->that(new ResideInOneOfTheseNamespaces('Sylius\Component'))
+            ->should(new NotDependsOnTheseNamespaces('Sylius\Component\Core'))
+            ->because('Core should not be used in any other component')
```

Comment:
```suggestion
            ->because('Core should not be used in stand-alone components')
```
So it will be in line with
> Core should not be used in stand-alone bundles